config/data_scrubber_test.go

```diff
@@ -0,0 +1,143 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func setupDataScrubber() *DataScrubber {
+	customSensitiveWords := []string{
+		"consul_token",
+		"dd_password",
+		"blocked_from_yaml",
+	}
+
+	scrubber := NewDefaultDataScrubber()
+	scrubber.SetCustomSensitiveWords(customSensitiveWords)
+
+	return scrubber
+}
+
+func TestBlacklistedArgs(t *testing.T) {
+	cases := []struct {
+		cmdline       []string
+		parsedCmdline []string
+	}{
+		{[]string{"agent", "-password", "1234"}, []string{"agent", "-password", "********"}},
+		{[]string{"agent", "--password", "1234"}, []string{"agent", "--password", "********"}},
+		{[]string{"agent", "-password=1234"}, []string{"agent", "-password=********"}},
+		{[]string{"agent", "--password=1234"}, []string{"agent", "--password=********"}},
+		{[]string{"fitz", "-consul_token=1234567890"}, []string{"fitz", "-consul_token=********"}},
+		{[]string{"fitz", "--consul_token=1234567890"}, []string{"fitz", "--consul_token=********"}},
+		{[]string{"fitz", "-consul_token", "1234567890"}, []string{"fitz", "-consul_token", "********"}},
+		{[]string{"fitz", "--consul_token", "1234567890"}, []string{"fitz", "--consul_token", "********"}},
+		{[]string{"python ~/test/run.py --password=1234 -password 1234 -open_password=admin -consul_token 2345 -blocked_from_yaml=1234 &"},
+			[]string{"python", "~/test/run.py", "--password=********", "-password", "********", "-open_password=admin", "-consul_token", "********", "-blocked_from_yaml=********", "&"}},
+		{[]string{"agent", "-PASSWORD", "1234"}, []string{"agent", "-PASSWORD", "********"}},
+		{[]string{"agent", "--PASSword", "1234"}, []string{"agent", "--PASSword", "********"}},
+		{[]string{"agent", "--PaSsWoRd=1234"}, []string{"agent", "--PaSsWoRd=********"}},
+		{[]string{"java -password      1234"}, []string{"java", "-password", "", "", "", "", "", "********"}},
+	}
+
+	scrubber := setupDataScrubber()
+	t.Log("scrub enabled: ", scrubber.Enabled)
+	t.Log("default regexp", scrubber.DefaultSensitiveWords)
+	t.Log("custom regexp", scrubber.CustomSensitiveWords)
+	t.Log("merged regexp", scrubber.SensitiveWords)
```

Comment:
I generally don't think we want logging in tests like this committed. If you need extra context for failures then it can go in the message arg of the `assert`